mesonbuild/utils/universal.py

```diff
@@ -1091,6 +1092,16 @@ def has_path_sep(name: str, sep: str = '/\\') -> bool:
     return False
 
 
+def flatten_path(name: str, sep: str = '/\\', replace: str = '_') -> str:
+    'Replaces any specified @sep path separators in @name with @replace'
+    result = ''
+    for char in name:
+        if char in sep:
+            char = replace
+        result = result + char
+    return result
```

Comment:
This is just `str.replace('/\\', '_')`, right?

Reply:
ohhh, never mind, this is relying on the iterability of `str`... Not sure how I feel about that.

Reply:
I'm a python novice, so if you have a more pythonic way to do this, I'd love to use it.

Reply:
Iterability of str is fine... how else would we do `if 's' in 'my string'` without treating it as a progression of individual one-char strings?

There's an unrelated topic where API surfaces that accept `Sequence[str]` in order to be generic across tuples or lists, also accepts raw strings because a str is a sequence of one-char strings, and that's not intended behavior for the API.

Reply:
My concern is that by using a string as the iterable you couldn’t have multi-character replacement. I’m thinking of windows sometimes using \\ in paths

Reply:
I'm not sure I've ever seen \\ in a windows path, but then I don't use windows often. In any case, this code uses the same logic as `has_path_sep` above, which looks only for single specific characters, not strings. Although, I guess you could pass a list of strings to `has_path_sep` while `flatten_path` really only allows strings. There aren't any uses of `has_path_sep` which even mention the optional `sep` arg, so it seems more theoretical than practical, I suspect.

Reply:
`\\` is a single character in a string; it's two characters in source code for escaping purposes only.

> This is just `str.replace('/\\', '_')`, right?

This is not equivalent; the first argument is searched verbatim, it is not a set of characters to be replaced as is done here.

Reply:
You probably want `re.sub(r'[\\/]', '_', name)`

Reply:
> You probably want `re.sub(r'[\\/]', '_', name)`

Good idea. However, I'd like to preserve the ABI similarity with `has_path_sep`, which means taking the set of to-be-replaced characters as a string argument. Constructing a regex that matches anything in a random string seems "hard" as it'd need to have characters special to regex escaped. Doing a bit of googling, this seems to work:

```
def flatten_path(name: str, sep: str = '/\\', replace: str = '_') -> str:
    '''Replaces any specified @sep path separators in @name with @replace'''
    return name.translate(name.maketrans(sep, replace * len(sep)))
```

Reply:
I've pushed this version after doing more testing